app/questionnaire/questionnaire_schema.py

```diff
@@ -94,11 +95,39 @@ def __init__(
         self._populate_answer_dependencies()
         self._populate_when_rules_section_dependencies()
         self._populate_calculated_summary_section_dependencies()
+        self._populate_min_max_map()
 
     @cached_property
     def answer_dependencies(self) -> ImmutableDict[str, set[AnswerDependent]]:
         return ImmutableDict(self._answer_dependencies_map)
 
+    def _populate_min_max_map(self) -> None:
+        for answer_id, answer in self._answers_by_id.items():
+            if (answer_type := answer[0].get("type")) and answer_type not in [
+                "Date",
+                "MonthYearDate",
+                "YearDate",
+            ]:
+                answer_to_search_for_min_max = self.get_answers_by_answer_id(answer_id)[
```

Comment:
Why are we searching for answers by id here when are already in the context of the answer from the loop above?

Reply:
+1